lib/browser/chrome-extension.js

```diff
@@ -69,74 +114,69 @@ app.on('will-quit', function () {
 
 // We can not use protocol or BrowserWindow until app is ready.
 app.once('ready', function () {
-  var chromeExtensionHandler, i, init, len, srcDirectory
-
   // Load persisted extensions.
   loadedExtensionsPath = path.join(app.getPath('userData'), 'DevTools Extensions')
   try {
-    loadedExtensions = JSON.parse(fs.readFileSync(loadedExtensionsPath))
-    if (!Array.isArray(loadedExtensions)) {
-      loadedExtensions = []
-    }
-
-    // Preheat the extensionInfo cache.
-    for (i = 0, len = loadedExtensions.length; i < len; i++) {
-      srcDirectory = loadedExtensions[i]
-      getExtensionInfoFromPath(srcDirectory)
+    let loadedExtensions = JSON.parse(fs.readFileSync(loadedExtensionsPath))
+    if (Array.isArray(loadedExtensions)) {
+      // Preheat the manifest cache.
+      for (let srcDirectory of loadedExtensions) {
+        getManifestFromPath(srcDirectory)
+      }
     }
   } catch (error) {
     // Ignore error
   }
 
   // The chrome-extension: can map a extension URL request to real file path.
-  chromeExtensionHandler = function (request, callback) {
-    var directory, parsed
-    parsed = url.parse(request.url)
-    if (!(parsed.hostname && (parsed.path != null))) {
-      return callback()
-    }
-    if (!/extension-\d+/.test(parsed.hostname)) {
-      return callback()
-    }
-    directory = getPathForHost(parsed.hostname)
-    if (directory == null) {
-      return callback()
+  const chromeExtensionHandler = function (request, callback) {
+    let parsed = url.parse(request.url)
+    if (!parsed.hostname || !parsed.path) return callback()
+    if (!/extension-\d+/.test(parsed.hostname)) return callback()
+
+    let directory = getPathForHost(parsed.hostname)
+    if (!directory) return callback()
+
+    if (parsed.path === '/_generated_background_page.html' &&
+        backgroundPages[parsed.hostname]) {
+      return callback({
+        mimeType: 'text/html',
+        data: backgroundPages[parsed.hostname].html
+      })
     }
-    return callback(path.join(directory, parsed.path))
+
+    fs.readFile(path.join(directory, parsed.path), function (err, content) {
+      if (err)
+        callback(-6)  // FILE_NOT_FOUND
+      else
+        return callback({mimeType: 'text/html', data: content})
```

Comment:
I left this comment originally since it seemed like @zcbenz was removing automatically inserted returns from the CoffeeScript days, see lines 157 and 171 for other examples and it seemed like this was another case looking at line 105 in the old diff.

I'm totally into being defensive but consistency, consistency, consistency...